build_byod_snapshot.sh

```diff
@@ -0,0 +1,319 @@
+        --query 'Credentials.[AccessKeyId,SecretAccessKey,SessionToken]' \
+        --output text
+    )
+    export AWS_ACCESS_KEY_ID=$ak
+    export AWS_SECRET_ACCESS_KEY=$sa
+    export AWS_SESSION_TOKEN=$st
+}
+
+assume_bootstrap_role() {
+    # to assume this role one should have credentials from anyscale-dev-product account
+    assume_role arn:aws:iam::830883877497:role/bk-premerge-first-jawfish-large-Role
+}
+
+check_role() {
+    aws sts get-caller-identity
+}
+
+setup_local_registry() {
+    # Run the local docker registry
+    print_info "Starting local Docker registry"
+    docker run -d -p $DOCKER_REGISTRY_PORT:5000 --restart always --name registry registry:2
+    print_info "Local Docker registry started, listening port $DOCKER_REGISTRY_PORT"
+}
+
+setup_tool_repo() {
+    pushd $TMP_REPO_DIR
+    print_info "Cloning ray-project/ray.git"
+    # TODO @timy remove this once dockerfile is placed in anyscale/templates repo
+    git clone git@github.com:ray-project/ray.git
+    print_info "Cloning anyscale/product.git"
+    git clone git@github.com:anyscale/product.git
+    popd # TMP_REPO_DIR
+}
+
+setup_crane() {
+    print_info "Downloading crane from ${CRANE_URL}"
+    curl -sL "${CRANE_URL}" | tar xzf - -C /tmp/
+    print_info "Copy crane binary to /usr/local/bin/"
+    sudo mv /tmp/crane /usr/local/bin/
+    print_info "Finish crane setup"
+}
+
+# Function to parse command-line arguments
+parse_args() {
+    # Require at least 4 arguments (2 options with 2 values)
+    if [ $# -lt 4 ]; then
+        echo "Usage: $0 --docker-file-path <path> --tag <tag>"
+        exit 1
+    fi
+
+    # Parse arguments
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            --docker-file-path)
+                docker_file_path="$2"
+                shift # Move past argument value
+                ;;
+            --tag)
+                tag="$2"
+                shift # Move past argument value
+                ;;
+            *)
+                echo "Unknown option: $1"
+                exit 1
+                ;;
+        esac
+        shift # Move to next argument
+    done
+
+    # Check if required arguments are set
+    if [ -z "$docker_file_path" ] || [ -z "$tag" ]; then
+        echo "Both --docker-file-path and --tag are required."
+        exit 1
+    fi
+
+    export DOCKER_FILE_PATH=$docker_file_path
+    export DOCKER_IMAGE_TAG=$tag
+}
+
+cleanup() {
+    print_info "Performing cleanup operations..."
+    print_info "cleanup $SNAPSHOTS_DIR"
+    rm -rf $SNAPSHOTS_DIR
+    print_info "cleanup $TMP_REPO_DIR"
+    rm -rf $TMP_REPO_DIR
+    print_info "cleanup local registry"
+    docker stop registry
+    docker rm registry
+}
+
+init_setup() {
+    parse_args "$@"
+    export SNAPSHOTS_DIR="$(mktemp -d -p /tmp)"
+    export TMP_REPO_DIR="$(mktemp -d -p /tmp)"
+    export CRANE_URL="https://github.com/google/go-containerregistry/releases/download/v0.15.1/go-containerregistry_Linux_x86_64.tar.gz"
```

Comment:
make these global and declared at the top with the default values so that advanced users can override them.